Q1: What is the difference between a t-test and a z-test? Provide an example scenario where you would use each type of test.
A t-test and a z-test are both statistical tests used to make inferences about population parameters based on sample data. However, there are some differences between the two tests, primarily related to the assumptions made about the population and the sample size.

A z-test assumes that the sample comes from a normally distributed population, and the population standard deviation is known. This test is used when the sample size is large (typically, n > 30) and the population variance is known or can be assumed to be known. For example, if a company wants to test if the mean height of their employees is different from the national average, and they know the standard deviation of the population height, they can use a z-test.

On the other hand, a t-test assumes that the sample comes from a normally distributed population, and the population standard deviation is unknown. This test is used when the sample size is small (typically, n < 30) or when the population variance is unknown. For example, if a researcher wants to test if a new drug has a different effect than a placebo, they can use a t-test to compare the mean scores of two groups of participants.

Example scenarios:

A company wants to compare the mean salary of their employees to the national average salary. They have a large sample size (n>30) and know the population standard deviation. They can use a z-test to make this comparison.

A researcher wants to compare the mean test scores of two groups of students. They have a small sample size (n<30) and do not know the population standard deviation. They can use a t-test to make this comparison.

Q2: Differentiate between one-tailed and two-tailed tests.
In hypothesis testing, the null hypothesis (H0) is a statement about a population parameter that is assumed to be true unless there is sufficient evidence to reject it in favor of an alternative hypothesis (Ha). The alternative hypothesis is the statement that the researcher is trying to support.

One-tailed tests and two-tailed tests are types of hypothesis tests that differ in the directionality of the alternative hypothesis.

A one-tailed test is a hypothesis test in which the alternative hypothesis specifies the direction of the effect or difference being tested. That is, the researcher is interested in whether the population parameter is either greater than or less than a certain value, but not both. Therefore, the rejection region for the null hypothesis is only on one side of the distribution. A one-tailed test is often used when the researcher has a specific directional hypothesis or when there is a theoretical or practical reason to expect an effect in a specific direction.

For example, a one-tailed test might be used to test the hypothesis that a new medication will decrease the amount of time it takes for patients to fall asleep. The alternative hypothesis would be that the medication decreases the amount of time it takes for patients to fall asleep, and the null hypothesis would be that the medication has no effect on the amount of time it takes for patients to fall asleep.

In contrast, a two-tailed test is a hypothesis test in which the alternative hypothesis does not specify the direction of the effect or difference being tested. That is, the researcher is interested in whether the population parameter is different from a certain value in any direction. Therefore, the rejection region for the null hypothesis is split between both sides of the distribution. A two-tailed test is often used when the researcher has no specific directional hypothesis or when there is no theoretical or practical reason to expect an effect in a specific direction.

For example, a two-tailed test might be used to test the hypothesis that the mean weight of male and female newborns is the same. The alternative hypothesis would be that the mean weight of male and female newborns is different, and the null hypothesis would be that the mean weight of male and female newborns is equal.

In summary, one-tailed tests and two-tailed tests differ in the directionality of the alternative hypothesis, and the choice between the two depends on the research question and the directional nature of the hypothesis.

Q3: Explain the concept of Type 1 and Type 2 errors in hypothesis testing. Provide an example scenario for each type of error.
In hypothesis testing, Type I and Type II errors are two types of errors that can occur when testing a hypothesis.

Type I error is the error that occurs when the null hypothesis is rejected when it is actually true. In other words, it is the false positive error, where we reject the null hypothesis even though it is correct. The probability of making a Type I error is denoted by alpha (α) and is often set at 0.05 or 0.01.

For example, suppose a researcher is testing the effectiveness of a new medication on a disease. The null hypothesis is that the medication has no effect on the disease, and the alternative hypothesis is that the medication has an effect. If the researcher rejects the null hypothesis and concludes that the medication is effective, but in reality, it is not, then a Type I error has occurred.

Type II error is the error that occurs when the null hypothesis is not rejected when it is actually false. In other words, it is the false negative error, where we fail to reject the null hypothesis even though it is incorrect. The probability of making a Type II error is denoted by beta (β).

For example, suppose a researcher is testing the effectiveness of a new medication on a disease. The null hypothesis is that the medication has no effect on the disease, and the alternative hypothesis is that the medication has an effect. If the researcher fails to reject the null hypothesis and concludes that the medication is not effective, but in reality, it is effective, then a Type II error has occurred.

The relationship between Type I and Type II errors is inverse, meaning that decreasing the probability of one type of error increases the probability of the other type of error.

Example scenario:

Type I error: A car alarm system is designed to go off if an intruder tries to break in. However, if the alarm goes off when no one is trying to break in, it is a false positive, which is a Type I error.

Type II error: A pregnancy test is designed to detect the presence of the hormone hCG, which indicates pregnancy. However, if the test does not detect hCG when the woman is actually pregnant, it is a false negative, which is a Type II error.

Q4: Explain Bayes's theorem with an example.
Bayes's theorem is a mathematical formula used to calculate the conditional probability of an event based on prior knowledge of related events. It is named after Thomas Bayes, an 18th-century statistician, and philosopher.

Bayes's theorem can be expressed as follows:

P(A|B) = P(B|A) * P(A) / P(B)

where P(A|B) is the probability of event A given event B has occurred, P(B|A) is the probability of event B given that event A has occurred, P(A) is the prior probability of event A, and P(B) is the prior probability of event B.

An example scenario where Bayes's theorem can be applied is in medical diagnosis. Suppose a patient exhibits certain symptoms that may indicate the presence of a disease. The probability of having the disease given the symptoms can be calculated using Bayes's theorem.

Let's say that the disease is present in 1% of the population, and the probability of a person exhibiting the symptoms given that they have the disease is 90%. On the other hand, the probability of a person exhibiting the symptoms without having the disease is 10%. Using this information, we can apply Bayes's theorem to calculate the probability of a person having the disease given that they exhibit the symptoms.

P(Disease|Symptoms) = P(Symptoms|Disease) * P(Disease) / P(Symptoms)

where P(Disease|Symptoms) is the probability of having the disease given the symptoms, P(Symptoms|Disease) is the probability of exhibiting the symptoms given the disease, P(Disease) is the prior probability of having the disease, and P(Symptoms) is the prior probability of exhibiting the symptoms.

Plugging in the values, we get:

P(Disease|Symptoms) = 0.90 * 0.01 / ((0.90 * 0.01) + (0.10 * 0.99))

= 0.08

Therefore, the probability of having the disease given the symptoms is 8%. This means that even though the patient exhibits the symptoms, there is still only an 8% chance of having the disease.

Q5: What is a confidence interval? How to calculate the confidence interval, explain with an example.
A confidence interval is a range of values that is likely to contain the true value of a population parameter with a certain degree of confidence. In other words, it is a range of values around a sample statistic that is likely to contain the true population parameter value with a certain level of certainty or probability.

To calculate a confidence interval, we need to know the sample mean, sample size, and standard deviation (or standard error) of the sample statistic. The formula for a confidence interval is:

Confidence interval = sample mean ± (z-score * standard error)

where z-score is the number of standard deviations corresponding to the desired level of confidence. For example, for a 95% confidence interval, the z-score would be 1.96.

Here's an example of how to calculate a confidence interval:

Suppose we want to estimate the average height of all people in a city. We take a random sample of 50 people and find that the sample mean height is 170 cm, and the sample standard deviation is 5 cm.

To calculate a 95% confidence interval for the population mean height, we need to first calculate the standard error. The standard error is calculated as:

Standard error = sample standard deviation / sqrt(sample size)

= 5 / sqrt(50)

= 0.71

Next, we can calculate the confidence interval as:

Confidence interval = 170 ± (1.96 * 0.71)

= 168.6 to 171.4 cm

This means that we can be 95% confident that the true population mean height falls within the range of 168.6 to 171.4 cm. We can interpret this as if we repeat this sampling process many times, 95% of the time, the true population mean height will be contained within this interval.

Q6. Use Bayes' Theorem to calculate the probability of an event occurring given prior knowledge of the event's probability and new evidence. Provide a sample problem and solution.
Sure, here's an example problem that uses Bayes' theorem:

Suppose a new medical test has been developed to detect a certain disease, which affects 1% of the population. The test has a false positive rate of 5%, meaning that 5% of the time, a healthy person will test positive for the disease. The test also has a false negative rate of 10%, meaning that 10% of the time, a person with the disease will test negative. Suppose a person takes the test and receives a positive result. What is the probability that the person actually has the disease?

Let's define some terms first:

A: the event that the person has the disease B: the event that the person tests positive for the disease We want to find P(A|B), the probability that the person has the disease given a positive test result. We can use Bayes' theorem to calculate this as follows:

P(A|B) = P(B|A) * P(A) / P(B)

where P(B|A) is the probability of testing positive given that the person has the disease, P(A) is the prior probability of the person having the disease, and P(B) is the probability of testing positive for the disease.

Using the information given in the problem, we can calculate these probabilities as follows:

P(B|A) = 1 - 0.10 = 0.90 (the probability of testing positive given that the person has the disease is 90%) P(A) = 0.01 (the prior probability of the person having the disease is 1%) P(B) = P(B|A) * P(A) + P(B|not A) * P(not A) = 0.90 * 0.01 + 0.05 * 0.99 (the probability of testing positive is the sum of the probability of a true positive and a false positive) Plugging these values into the Bayes' theorem formula, we get:

P(A|B) = 0.90 * 0.01 / (0.90 * 0.01 + 0.05 * 0.99)

= 0.15

Therefore, the probability that the person actually has the disease given a positive test result is 15%. This means that the test is not very accurate, and there is still a high chance of a false positive result.

Question 7 : Calculate the 95% confidence interval for a sample of data with a mean of 50 and a standard deviation of 5. Interpret the results.
Answer :
Sample Size is not provided in above Question , I am assuming sample size of 30

In [1]:
import math
import scipy.stats as stats

# Define the sample mean, standard deviation, and sample size
sample_mean = 50
sample_std_dev = 5
sample_size = 30

# Signficance Level
alpha = 0.05

# Calculate the critical value for a 95% confidence interval
t_crit = stats.t.ppf(1-alpha/2, df=sample_size-1)
print(f't statistic for significance level {alpha} , sample size of {sample_size} is : {t_crit:.2f}')

# Calculate the standard error
std_error = sample_std_dev / math.sqrt(sample_size)

# Calculate the lower and upper bounds of the confidence interval
lower_bound = sample_mean - t_crit * std_error
upper_bound = sample_mean + t_crit * std_error

# Print the results
print(f"The 95% confidence interval is ({lower_bound:.2f}, {upper_bound:.2f})")

t statistic for significance level 0.05 , sample size of 30 is : 2.05
The 95% confidence interval is (48.13, 51.87)


Above results mean that I can say with 95% condfidence that the POPULATION MEAN Lies in between 48.13 and 51.87

Question 8 : What is the margin of error in a confidence interval? How does sample size affect the margin of error?
Provide an example of a scenario where a larger sample size would result in a smaller margin of error


Answer :

The margin of error in a confidence interval is the range of values around the sample statistic (such as the sample mean) within which we expect the true population parameter (such as the population mean) to lie with a certain level of confidence. The margin of error is affected by the sample size, level of confidence, and standard deviation of the population.

As the sample size increases, the margin of error decreases because larger samples provide more information about the population, and hence the sample statistics are more likely to be representative of the population. The decrease in margin of error with an increase in sample size follows the square root law, which states that the margin of error decreases by the square root of the sample size.

For example, suppose we want to estimate the average height of all students in a university using a random sample of students. We take a sample of 50 students and calculate their average height to be 170 cm with a standard deviation of 5 cm. We want to calculate a 95% confidence interval for the true population mean height.

Using the formula for the margin of error for a confidence interval, we get:

Margin of error = z * (standard deviation / sqrt(sample size))

For a 95% confidence interval, the critical z-value is 1.96. So, plugging in the values, we get:

Margin of error = 1.96 * (5 / sqrt(50)) = 1.38 cm

This means that we are 95% confident that the true population mean height is within 1.38 cm of our sample mean of 170 cm. If we had taken a larger sample size of 200 students, the margin of error would have been:

Margin of error = 1.96 * (5 / sqrt(200)) = 0.98 cm

So, a larger sample size would result in a smaller margin of error, making our estimate of the true population mean height more precise.

Question 9 : Calculate the z-score for a data point with a value of 75, a population mean of 70, and a population standard deviation of 5. Interpret the results.

Answer :
Below is my python code to calculate above Z Score

In [2]:
# Define the values
x = 75  # Data point value
mu = 70  # Population mean
sigma = 5  # Population standard deviation

# Calculate the z-score
z = (x - mu) / sigma

print(f"The z-score is: {z}")

The z-score is: 1.0


This means that the data point is 1 standard deviation above the population mean. A positive z-score indicates that the data point is above the mean, while a negative z-score indicates that it is below the mean. The magnitude of the z-score tells us how far away the data point is from the mean in terms of standard deviations.

In [3]:
from scipy.stats import norm

# Calculate the percentage of values below a z-score of 1
percent_below = norm.cdf(z) * 100

print(f"The percentage of values below a z-score of 1 is: {percent_below:.2f}%")

The percentage of values below a z-score of 1 is: 84.13%


A z-score of 1 corresponds to approximately 84.13% of values being below this value in a standard normal distribution.

Question 10 : In a study of the effectiveness of a new weight loss drug, a sample of 50 participants lost an average of 6 pounds with a standard deviation of 2.5 pounds. Conduct a hypothesis test to determine if the drug is significantly effective at a 95% confidence level using a t-test.

Answer :

In [4]:
import numpy as np
from scipy.stats import t

# Null hypothesis: the drug is not significantly effective
# Alternative hypothesis: the drug is significantly effective
alpha = 0.05  # significance level
null_hypothesis = "the drug is not significantly effective"
alternative_hypothesis = "the drug is significantly effective"

mu = 0
sample_mean = 6
sample_std = 2.5
n = 50
df = n - 1  # degrees of freedom

# Calculate the t-score and p-value
t_score = (sample_mean - mu) / (sample_std / np.sqrt(n))
p_value = 2 * (1 - t.cdf(abs(t_score), df))

# Compare p-value with alpha and make a conclusion
print(f"t-statistic: {t_score:.4f}")
print(f"p-value: {p_value}")
if p_value < alpha:
    print(f"Reject the null hypothesis. {alternative_hypothesis}.")
else:
    print(f"Fail to reject the null hypothesis. {null_hypothesis}.")

t-statistic: 16.9706
p-value: 0.0
Reject the null hypothesis. the drug is significantly effective.


Question 11 : In a survey of 500 people, 65% reported being satisfied with their current job. Calculate the 95% confidence interval for the true proportion of people who are satisfied with their job.

Answer :

In [6]:
import math

p = 0.65  # sample proportion
n = 500  # sample size
z_alpha = 1.96  # z-score for 95% confidence level

# Calculate the standard error
se = math.sqrt((p * (1 - p)) / n)

# Calculate the margin of error
me = z_alpha * se

# Calculate the confidence interval
lower_bound = p - me
upper_bound = p + me

# Print the results
print(f"The 95% confidence interval for the proportion of people who are satisfied with their job is {lower_bound*100:.2f}%")

The 95% confidence interval for the proportion of people who are satisfied with their job is 60.82%


Question 12 : A researcher is testing the effectiveness of two different teaching methods on student performance.Sample A has a mean score of 85 with a standard deviation of 6, while sample B has a mean score of 82 with a standard deviation of 5. Conduct a hypothesis test to determine if the two teaching methods have a significant difference in student performance using a t-test with a significance level of 0.01.

Answer :

In [7]:
import numpy as np
from scipy.stats import t

# Sample A
x1 = 85
s1 = 6
n1 = 30

# Sample B
x2 = 82
s2 = 5
n2 = 30

# Set up null and alternative hypotheses
# H0: mu1 = mu2 (the means are equal)
# Ha: mu1 != mu2 (the means are not equal)
alpha = 0.01
null_hypothesis = "mu1 = mu2 (the means are equal)"
alternative_hypothesis = "mu1 != mu2 (the means are not equal)"

# Calculate pooled standard deviation
sp = np.sqrt(((n1 - 1) * s1 ** 2 + (n2 - 1) * s2 ** 2) / (n1 + n2 - 2))

# Calculate t-statistic
t_stat = (x1 - x2) / (sp * np.sqrt(1 / n1 + 1 / n2))

# Calculate degrees of freedom
df = n1 + n2 - 2

# Calculate critical t-value
t_crit = t.ppf(1-alpha / 2, df)

# Calculate p-value
p_value = 2 * (1 - t.cdf(abs(t_stat), df))

# Print results
print(f"t-statistic: {t_stat:.3f}")
print(f"Degrees of freedom: {df}")
print(f"Critical t-value: {t_crit:.3f}")
print(f"p-value: {p_value:.3f}")

if abs(t_stat) > t_crit:
    print("Reject null hypothesis.")
    print(alternative_hypothesis)
else:
    print("Fail to reject null hypothesis.")
    print(null_hypothesis)

t-statistic: 2.104
Degrees of freedom: 58
Critical t-value: 2.663
p-value: 0.040
Fail to reject null hypothesis.
mu1 = mu2 (the means are equal)


Q13. A population has a mean of 60 and a standard deviation of 8. A sample of 50 observations has a mean of 65. Calculate the 90% confidence interval for the true population mean.

In [8]:
import math
from scipy.stats import t

pop_mean = 60
pop_std = 8
n = 50
sample_mean = 65
confidence_level = 0.90

# Calculate the t-score
t_score = (sample_mean - pop_mean) / (pop_std / math.sqrt(n))

# Find the critical t-value
df = n - 1
t_crit = t.ppf((1+confidence_level)/2, df)
print(f't statistic for {n} samples with {confidence_level*100:.0f}% confidence is : {t_crit:.4f}')

# Calculate the margin of error
margin_of_error = t_crit * (pop_std / math.sqrt(n))

# Calculate the confidence interval
lower_bound = sample_mean - margin_of_error
upper_bound = sample_mean + margin_of_error

# Print the results
print(f"The {confidence_level*100:.0f}% confidence interval for the true population mean is ({lower_bound:.3f}, {upper_bound:.4f})")

t statistic for 50 samples with 90% confidence is : 1.6766
The 90% confidence interval for the true population mean is (63.103, 66.8968)


Q14. In a study of the effects of caffeine on reaction time, a sample of 30 participants had an average reaction time of 0.25 seconds with a standard deviation of 0.05 seconds. Conduct a hypothesis test to determine if the caffeine has a significant effect on reaction time at a 90% confidence level using a t-test.

Answer :

In this example popluation mean is not provided
According to researchers typical human reaction time is 200ms to 300ms
Here i have assumed the population mean as 300ms or 0.3 seconds

In [10]:
import math
import scipy.stats as stats

# Null Hypothesis: Caffeine has no significant effect on reaction time
# Alternative Hypothesis: Caffeine has a significant effect on reaction time
null_hypothesis = "Caffeine has no significant effect on reaction time"
alternative_hypothesis = "Caffeine has a significant effect on reaction time"

n=30
sm=0.25
sd=0.05
popm=0.3
alpha=0.1
t_stat=(sm-popm)/(sd/math.sqrt(n))
t_crit=stats.t.ppf(1-alpha/2,n-1)
# Calculate the confidence interval
margin_of_error = t_crit * (sd / math.sqrt(n))
lower_ci = sm - margin_of_error
upper_ci = sm + margin_of_error

# Print the results
print(f"t-statistic: {t_stat:.3f}")
print(f"t-critical value: {t_crit:.3f}")
print(f"90% Confidence Interval: ({lower_ci:.3f}, {upper_ci:.3f})")

# Determine if the null hypothesis should be rejected or not
if abs(t_stat) > t_crit:
    print("Reject the null hypothesis")
    print(alternative_hypothesis)
else:
    print("Fail to reject the null hypothesis")
    print(null_hypothesis)



t-statistic: -5.477
t-critical value: 1.699
90% Confidence Interval: (0.234, 0.266)
Reject the null hypothesis
Caffeine has a significant effect on reaction time
